core/src/task.cpp

```diff
@@ -92,6 +92,7 @@ const ContainerBase* TaskPrivate::stages() const {
 
 Task::Task(const std::string& ns, bool introspection, ContainerBase::pointer&& container)
   : WrapperBase(new TaskPrivate(this, ns), std::move(container)) {
+	setPruning(ENABLE_PRUNING);
```

Comment:
Wouldn't it be possible to just have a C++ API with defaulting to true without exposing this via CMake?